### PR TITLE
Add users docs

### PR DESCRIPTION
**Pull request by hishamco**

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Users.Abstractions/Events/ILoginFormEvent.cs [hunk 1]
```diff
@@ -3,12 +3,28 @@
 
 namespace OrchardCore.Users.Events
 {
+    /// <summary>
+    /// Contract for login events.
+    /// </summary>
     public interface ILoginFormEvent
     {
+        /// <summary>
+        /// Occurs when the user is login.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Users.Abstractions/Events/ILoginFormEvent.cs]: logging

File: src/OrchardCore/OrchardCore.Users.Abstractions/Events/ILoginFormEvent.cs [hunk 2]
```diff
@@ -3,12 +3,28 @@
...
+        /// </summary>
+        /// <param name="userName">The username.</param>
+        /// <param name="reportError">The reported error if fauilar happened during the login process.</param>
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Users.Abstractions/Events/ILoginFormEvent.cs]: fauilar?

File: src/OrchardCore/OrchardCore.Users.Abstractions/Events/ILoginFormEvent.cs [hunk 3]
```diff
@@ -3,12 +3,28 @@
...
         Task LoggingInAsync(string userName, Action<string, string> reportError);
 
+        /// <summary>
+        /// Occurs when the user login is failed.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Users.Abstractions/Events/ILoginFormEvent.cs]: is failed => fails or has failed

File: src/OrchardCore/OrchardCore.Users.Abstractions/Events/IPasswordRecoveryFormEvents.cs [hunk 4]
```diff
@@ -5,14 +5,31 @@
 
 namespace OrchardCore.Users.Events
 {
+    /// <summary>
+    /// Contract for password recovery events.
+    /// </summary>
     public interface IPasswordRecoveryFormEvents
     {
+        /// <summary>
+        /// Occures during the user password resetting.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Users.Abstractions/Events/IPasswordRecoveryFormEvents.cs]: Occures => Occurs
resetting => is being reset

File: src/OrchardCore/OrchardCore.Users.Abstractions/Events/IPasswordRecoveryFormEvents.cs [hunk 5]
```diff
@@ -5,14 +5,31 @@
...
+        /// </summary>
+        /// <param name="reportError">The reported error if fauilar happened during the ressting process.</param>
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Users.Abstractions/Events/IPasswordRecoveryFormEvents.cs]: fauilar ?
ressting process => reset

File: src/OrchardCore/OrchardCore.Users.Abstractions/Events/IPasswordRecoveryFormEvents.cs [hunk 6]
```diff
@@ -5,14 +5,31 @@
...
         Task ResettingPasswordAsync(Action<string, string> reportError);
 
+        /// <summary>
+        /// Occures after the user password has been reset.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Users.Abstractions/Events/IPasswordRecoveryFormEvents.cs]: Replace every Occures with Occurs

File: src/OrchardCore/OrchardCore.Users.Abstractions/Handlers/IExternalLoginEventHandler.cs [hunk 7]
```diff
@@ -5,12 +5,22 @@
...
+        /// <summary>
+        /// Occurs when the username is generated.
+        /// </summary>
+        /// <param name="provider">The login provider.</param>
+        /// <param name="claims">A <see cref="SerializableClaim"/>s.</param>
+        /// <return>A username.</return>
         Task<string> GenerateUserName(string provider, IEnumerable<SerializableClaim> claims);
 
+        /// <summary>
+        /// Occurs when the user roles has been updated.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Users.Abstractions/Handlers/IExternalLoginEventHandler.cs]: user roles has been updated => user roles are updated

File: src/OrchardCore/OrchardCore.Users.Abstractions/Handlers/UpdateRolesContext.cs [hunk 8]
```diff
@@ -3,23 +3,48 @@
...
+        /// </summary>
         public string LoginProvider { get; }
 
+        /// <summary>
+        /// Gets a list of external claims.
+        /// </summary>
         public IEnumerable<SerializableClaim> ExternalClaims { get; }
 
+        /// <summary>
+        /// Gets a user roles.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Users.Abstractions/Handlers/UpdateRolesContext.cs]: a user roles => the user's roles

File: src/OrchardCore/OrchardCore.Users.Abstractions/Services/IUserService.cs [hunk 9]
```diff
@@ -5,18 +5,81 @@
 namespace OrchardCore.Users.Services
 {
     /// <summary>
-    /// This interface provides an abstraction for common user operations.
+    /// Contract that provides a common user operations.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Users.Abstractions/Services/IUserService.cs]: 'an abstraction for common user operations' was good.

**Comment by hishamco** [comment]
Thanks Antoine, I will fix them all after I will finish my investigation from another PR

File: src/OrchardCore/OrchardCore.Users.Abstractions/Events/IRegistrationFormEvents.cs [hunk 10]
```diff
@@ -3,10 +3,21 @@
...
 {
+    /// <summary>
+    /// Contract for user registration events.
+    /// </summary>
     public interface IRegistrationFormEvents
     {
+        /// <summary>
+        /// Occurs during the user registration.
+        /// </summary>
+        /// <param name="reportError">The reported error if fauilar happened in validation process.</param>
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Users.Abstractions/Events/IRegistrationFormEvents.cs]: fauilar => replace by failure everywhere

**Comment by hishamco** [comment]
Fixed!!

**Comment by agriffard** [comment]
Conflicts to solve.

**Pull request merged**